src/Lumi/Components/Form/Table.purs

```diff
@@ -68,30 +72,53 @@ instance applicativeTableFormBuilder :: Applicative (TableFormBuilder props row)
       , validate: \_ -> pure a
       }
 
+-- | Revalidate the table form, in order to display error messages or create
+-- | a validated result.
+revalidate
+  :: forall props row result
+   . TableFormBuilder props row result
+  -> props
+  -> row
+  -> Maybe result
+revalidate form props row = (un TableFormBuilder form props).validate row
+
 -- | A `TableFormBuilder` makes a `FormBuilder` for an array where each row has
 -- | columns defined by it.
 editableTable
   :: forall props row result
    . { addLabel :: String
-     , defaultValue :: Maybe row
+     , addRow :: Maybe (Aff (Maybe row))
```

Comment:
I think this could do with a comment to say what the meaning of each Maybe is. I guess the outer maybe tells the component whether to render the "Add row" link at all, and the inner one allows you to decide whether or not to actually add the row when the link is clicked?

Reply:
Good point, I'll add a docstring here. Yes, that's exactly what those types mean.